template/server/messaging.py

```diff
@@ -248,20 +280,45 @@ async def execute(
         env_vars: Dict[StrictStr, str] = None,
     ):
         message_id = str(uuid.uuid4())
-        logger.debug(f"Sending code for the execution ({message_id}): {code}")
-
         self._executions[message_id] = Execution()
 
         if self._ws is None:
             raise Exception("WebSocket not connected")
 
         async with self._lock:
-            # set env vars (will override global env vars)
+            # Wait for any pending cleanup task to complete
+            if self._cleanup_task and not self._cleanup_task.done():
+                logger.debug("Waiting for pending cleanup task to complete")
+                try:
+                    await self._cleanup_task
+                except Exception as e:
+                    logger.warning(f"Cleanup task failed: {e}")
+                finally:
+                    self._cleanup_task = None
+            
+            # Get the indentation level from the code
+            code_indent = self._get_code_indentation(code)
+            
+            # Build the complete code snippet with env vars
+            complete_code = code
+            env_var_snippets = []
```

Comment:
why do you snippets and not only env vars and transform them together?

Reply:
i was having an issue where the user env var snippets were set before global, but will redo this